libs/storage/Tsavorite/cs/src/core/TsavoriteLog/TsavoriteLogIterator.cs

```diff
@@ -472,6 +470,20 @@ public unsafe bool TryBulkConsumeNext<T>(T consumer, int maxChunkSize = 0) where
             {
                 while (true)
                 {
+                    // If initializing wait for completion
+                    while (tsavoriteLog.Initializing)
+                    {
+                        Thread.Yield();
+                        try
```

Comment:
Why not just do epoch.ProtectAndDrain() instead of suspend-resume?

Reply:
Addressed this by using epoch.ProtectAndDrain()